fe/fe-core/src/main/java/org/apache/doris/resource/resourcegroup/ResourceGroup.java

```diff
@@ -0,0 +1,123 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.resource.resourcegroup;
+
+import org.apache.doris.common.FeConstants;
+import org.apache.doris.common.io.DeepCopy;
+import org.apache.doris.common.io.Text;
+import org.apache.doris.common.io.Writable;
+import org.apache.doris.common.proc.BaseProcResult;
+import org.apache.doris.persist.gson.GsonUtils;
+import org.apache.doris.thrift.TPipelineResourceGroup;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.google.gson.annotations.SerializedName;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+
+public class ResourceGroup implements Writable {
+
+    private static final Logger LOG = LogManager.getLogger(ResourceGroup.class);
+
+    @SerializedName(value = "id")
+    private long id;
+
+    @SerializedName(value = "name")
+    private String name;
+
+    @SerializedName(value = "properties")
+    private Map<String, String> properties;
+
+    // version update required after alter resource group
+    @SerializedName(value = "version")
+    private long version;
+
+    @SerializedName(value = "parentName")
+    private String parentName;
+
+    @SerializedName(value = "childrenNames")
```

Comment:
Maybe using id is better?
The name will be more appropriate for user presentations.

Reply:
If use id here would have to maintain a map of idToResourceGroup in fe, which would get complicated. Here using name makes it easier to operate with nameToResourceGroup. But on the be side we can construct idToResourceGroup from list<TPipelineResourceGroup>, which is child to parent ordered.